Data/ByteString/Builder/Internal.hs

```diff
@@ -414,6 +418,12 @@ instance Monoid Builder where
   {-# INLINE mconcat #-}
   mconcat = foldr mappend mempty
 
+instance IsList Builder where
+  type Item Builder = Word8
+  fromList = lazyByteStringInsert . fromList
+  fromListN n = byteString . S.unsafePackLenBytes n
```

Comment:
`unsafePackLenBytes` is not appropriate here, since that will happily write memory out-of-bounds (or leave memory uninitialized) and this instance is usable via "safe" modules.

Come to think of it, this approach generally requires traversing the entire list before any output can be produced. Perhaps a better a pproach is `lazyByteString . fromListN n`, combined with making the `IsString LazyByteString` instance take advantage of a size hint?

Reply:
Assuming you mean `IsList LazyByteString`

Reply:
The only functions that I see that would work for this are `unsafePackLenBytes` and `packUptoLenBytes` so I'll use packUptoLenBytes and see if that works for you.

Reply:
> Also, the memory and performance benefit of using LazyByteString is questionable to me and if performance is essential I think that there are better options than fromListN.

I suppose that's true, at least while these instances are not good list consumers. 

> I do not know how to have fromListN for LazyByteString take advantage of the size parameter without using StrictByteString.

I had in mind floating out the `packChunks` helper function in `packBytes` and passing the size hint to the former. But that can wait for a follow-up PR if it's not needed in this one.